pkg/services/ngalert/ngalert.go

```diff
@@ -289,6 +289,24 @@ func (ng *AlertNG) init() error {
 	ng.AlertsRouter = alertsRouter
 
 	evalFactory := eval.NewEvaluatorFactory(ng.Cfg.UnifiedAlerting, ng.DataSourceCache, ng.ExpressionService, ng.pluginsStore)
+
+	var recordingWriter writer.Writer
+	if ng.FeatureToggles.IsEnabledGlobally(featuremgmt.FlagGrafanaManagedRecordingRules) {
+		promWriterCfg := writer.PrometheusWriterConfig{
+			URL:               ng.Cfg.UnifiedAlerting.RecordingRules.URL,
+			BasicAuthUsername: ng.Cfg.UnifiedAlerting.RecordingRules.BasicAuthUsername,
+			BasicAuthPassword: ng.Cfg.UnifiedAlerting.RecordingRules.BasicAuthPassword,
+			CustomHeaders:     ng.Cfg.UnifiedAlerting.RecordingRules.CustomHeaders,
+			Timeout:           10 * time.Second,
+		}
+		recordingWriter, err = writer.NewPrometheusWriter(promWriterCfg, log.New("ngalert.writer"))
+		if err != nil {
+			return err
+		}
+	} else {
+		recordingWriter = writer.NoopWriter{}
+	}
```

Comment:
Nit: Extract to separate function that returns (interface, err). Then you can drop `var` declaration and `else` block in favor of just returning. these having to exist are a smell that this should be extracted

Reply:
Fair point, this gives us a decent enough excuse to move away from the `writer.Writer` interface in this PR as well.